processor/otel/metadata.go

```diff
@@ -222,6 +226,32 @@ func ifaceAnyValueArray(array pdata.AnyValueArray) []interface{} {
 
 // initEventLabels initializes an event-specific label map, either making a copy
 // of commonLabels if it is non-nil, or otherwise creating a new map.
-func initEventLabels(commonLabels common.MapStr) common.MapStr {
-	return commonLabels.Clone()
+func initEventLabels(e *model.APMEvent) {
+	e.Labels = e.Labels.Clone()
+	e.NumericLabels = e.NumericLabels.Clone()
+}
+
+func setLabel(key string, event *model.APMEvent, v interface{}) {
+	switch v := v.(type) {
+	case string:
+		event.Labels[key] = v
+	case bool:
+		event.Labels[key] = strconv.FormatBool(v)
+	case float64:
+		event.NumericLabels[key] = v
+	case int64:
+		event.NumericLabels[key] = float64(v)
+	case []interface{}:
+		if len(v) == 0 {
+			return
+		}
+		switch v[0].(type) {
+		case string:
+			event.Labels[key] = v
+		case float64:
+			event.NumericLabels[key] = v
+		}
+	default:
+		panic(fmt.Sprintf("Unhandled %T, %+v\n", v, v))
```

Comment:
I'd rather we didn't panic in this code. If anything, just log a warning. We should probably remove some of the reflection at some point.